test/extended/images/resolve.go

```diff
@@ -128,6 +132,325 @@ var _ = g.Describe("[Feature:ImageLookup][registry][Conformance] Image policy",
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "resolve-template",
+			},
+			Spec: kbatchv1.JobSpec{
+				Template: kapiv1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							"alpha.image.policy.openshift.io/resolve-names": "*",
+						},
+					},
+					Spec: kapiv1.PodSpec{
+						TerminationGracePeriodSeconds: &one,
+						RestartPolicy:                 kapiv1.RestartPolicyNever,
+						Containers: []kapiv1.Container{
+							{Name: "resolve", Image: "busybox:latest", Command: []string{"/bin/true"}},
+						},
+					},
+				},
+			},
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(job.Spec.Template.Spec.Containers[0].Image).To(o.Equal(tag.Image.DockerImageReference))
+
+		g.By("auto replacing local references on CronJobs")
+		cronjob, err := oc.KubeClient().BatchV1beta1().CronJobs(oc.Namespace()).Create(&kbatchv1beta1.CronJob{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "resolve",
+				Annotations: map[string]string{
+					"alpha.image.policy.openshift.io/resolve-names": "*",
+				},
+			},
+			Spec: kbatchv1beta1.CronJobSpec{
+				Schedule: "1 0 * * *",
+				JobTemplate: kbatchv1beta1.JobTemplateSpec{
+					Spec: kbatchv1.JobSpec{
+						Template: kapiv1.PodTemplateSpec{
+							Spec: kapiv1.PodSpec{
+								TerminationGracePeriodSeconds: &one,
+								RestartPolicy:                 kapiv1.RestartPolicyNever,
+								Containers: []kapiv1.Container{
+									{Name: "resolve", Image: "busybox:latest", Command: []string{"/bin/true"}},
+								},
+							},
+						},
+					},
+				},
+			},
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		// FIXME: this needs actually fixing the admission first
+		o.Expect(cronjob.Spec.JobTemplate.Spec.Template.Spec.Containers[0].Image).NotTo(o.BeEmpty())
+		// o.Expect(cronjob.Spec.JobTemplate.Spec.Template.Spec.Containers[0].Image).To(o.Equal(tag.Image.DockerImageReference))
+
+		g.By("auto replacing local references on CronJobs (in pod template)")
+		cronjob, err = oc.KubeClient().BatchV1beta1().CronJobs(oc.Namespace()).Create(&kbatchv1beta1.CronJob{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "resolve-template",
+			},
+			Spec: kbatchv1beta1.CronJobSpec{
+				Schedule: "1 0 * * *",
+				JobTemplate: kbatchv1beta1.JobTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							"alpha.image.policy.openshift.io/resolve-names": "*",
+						},
+					},
+					Spec: kbatchv1.JobSpec{
+						Template: kapiv1.PodTemplateSpec{
+							Spec: kapiv1.PodSpec{
+								TerminationGracePeriodSeconds: &one,
+								RestartPolicy:                 kapiv1.RestartPolicyNever,
+								Containers: []kapiv1.Container{
+									{Name: "resolve", Image: "busybox:latest", Command: []string{"/bin/true"}},
+								},
+							},
+						},
+					},
+				},
+			},
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		// FIXME: this needs actually fixing the admission first
```

Comment:
v1beta1.cronjobs are missing in the switch in admission

Reply:
Right, that's why we need to the bump from https://github.com/openshift/apiserver-library-go/pull/23

Reply:
yep, I have reinstantiated the test after adding the fake bump